File: src/main/java/com/zyd/core/objecthandler/House.java
package com.zyd.core.objecthandler;

import java.util.HashMap;
import java.util.HashSet;

import com.tj.common.CommonUtil;
import com.zyd.core.db.HibernateUtil;
import com.zyd.core.util.Ocr;

@SuppressWarnings("unchecked")
public class House extends Handler {
    public final static String EntityName = HibernateUtil.EntityNames.House;
    private final static HashSet CDataColumns = new HashSet();

    public String getEntityName() {
        return EntityName;
    }

    @Override
    protected boolean beforeCreate(HashMap values) {
        String tel = (String) values.get(Columns.Tel);
        if (tel != null && tel.length() > 100) {
            String type = CommonUtil.getFileSuffix((String) values.get(Columns.TelImageName));
            values.put(Columns.Tel, Ocr.ocrImageNumber(tel, type));
        }
        return true;
    }

    public SearchResult query(HashMap params) {
        SearchResult r = super.query(params);
        r.cdataColumns = CDataColumns;
        return r;
    }

    public final static class Columns extends Handler.Columns {
        public final static String RentalType = "rentalType";
        public final static String SubRentalType = "subRentalType";
        public final static String Price = "price";
        public final static String PaymentType = "paymentType";
        public final static String PriceUit = "priceUit";
        public final static String Size = "size";
        public final static String HouseType = "houseType";
        public final static String Address = "address";
        public final static String District1 = "district1";
        public final static String District3 = "district3";
        public final static String District5 = "district5";
        public final static String Tel = "tel";
        public final static String Contact = "contact";
        public final static String Photo = "photo";
        public final static String Description1 = "description1";
        public final static String Description2 = "description2";
        public final static String Floor = "floor";
        public final static String TotalFloor = "totalFloor";
        public final static String IsAgent = "isAgent";
        public final static String Equipment = "equipment";
        public final static String Decoration = "decoration";
        public final static String TelImageName = "telImageName";
    }

}

File: src/main/webapp/js/gui/house.js
var pageSize = 50;
var columns = [  'lng', 'lat', 'price', 'id','rentalType', 'subRentalType',  'paymentType', 'priceUnit', 'size', 'houseType', 'address', 'city',
        'district1', 'district3', 'district5', 'tel', 'contact', 'photo', 'floor', 'totalFloor', 'isAgent', 'agentPhoto',
        'equipment', 'decoration', 'ok', 'referer', 'createTime', 'updateTime', 'hash' , 'description1', 'description2'];

var defaultParams = {
    objectid : 'House',
    count : pageSize
};

function loadit() {
    var store = new Ext.data.Store( {
        autoLoad : false,
        remoteSort : true,
        url : '/service/object',
        restful : true,
        baseParams : defaultParams,
        reader : new Ext.data.XmlReader( {
            record : 'object',
            id : 'id'
        }, columns),
        paramNames : {
            start : 'start',
            limit : 'count',
            sort : 'orderBy',
            dir : 'order'
        },
        getTotalCount : function() {
            return 9000000;
        }
    });

    var headers = [];
    for ( var i = 0; i < columns.length; i++) {
        var c = columns[i];
        headers.push( {
            header : c,
            dataIndex : c,
            sortable : true
        });
    }
    var grid = new Ext.grid.GridPanel( {
        store : store,
        columns : headers,
        bbar : new Ext.PagingToolbar( {
            pageSize : pageSize,
            store : store,
             displayInfo : true,
            displayMsg : 'Displaying  {0} - {1} of {2}',
            emptyMsg : "Nothing to display"      
        }),
        sm : new Ext.grid.RowSelectionModel( {
            singleSelect : true,
            listeners : {
                rowselect : function(sm, row, rec) {
                    Ext.getCmp("form1").getForm().loadRecord(rec);
                    var photo = rec.data.photo, agentPhoto=rec.data.agentPhoto;                    
                    var imagePanel = Ext.getCmp("imagePanel");                    
                    imagePanel.removeAll(true);
                    if(!photo && !agentPhoto){                    
                        imagePanel.collapse();
                        return;
                    }
                    
                    imagePanel.expand();                    
                    var elements = [];       
                    if(agentPhoto && agentPhoto.length!=0){
                        elements.push({xtype:'panel', html:'<img src="/service/image?name='+agentPhoto+'">'});
                    }
                    if(photo && photo.length!=0){
                        photo = photo.split(';');
                        for(var i=0;i<photo.length;i++){
                            elements.push({xtype:'panel', html:'<img src="/service/image?name='+photo[i]+'">'});
                        }
                    }                    
                    imagePanel.add(elements);
                    imagePanel.doLayout();
                    
                }
            }
        })

    });
    var fields = [];
    for ( var i = 0; i < columns.length; i++) {
        var c = columns[i];
        fields.push( {
            fieldLabel : c,
            name : c
        });
    }
    fields[fields.length-1].xtype='textarea';
    fields[fields.length-1].width =160;
    fields[fields.length-1].height =100;
    fields[fields.length-2].xtype='textarea';
    fields[fields.length-2].width =160;
    fields[fields.length-2].height =80;
    
    var form = new Ext.FormPanel( {
        id : 'form1',
        frame : true,
        defaultType : 'textfield',
        labelWidth : 90,
        width : 300,                
        items : fields,
        autoScroll:true,
        buttons : [ {
            text : 'Search',
            handler : function(button, event) {
                var values = form.getForm().getValues(), param={};
                Ext.apply(param, defaultParams);
                Ext.apply(param, values);
                store.baseParams = param;
                store.load();                
            }
        } ]
    });
    
    var imagePanel = new Ext.Panel({
        id:'imagePanel',
        width:300,
        height:'100%',
        autoScroll:true,        
        collapsed:true
    });
        
    
    grid.region = 'center';
    form.region = 'west';
    imagePanel.region = 'east';
    
    form.collapsible = true;
    form.split = true;
    
    imagePanel.collapsible = true;
    imagePanel.split = true;
    
    new Ext.Viewport( {
        layout : 'border',
        items : [ grid, form, imagePanel ]
    });

    store.load( {
        params : {
            start : 0            
        }
    });
}


function validQueryParam(params){
    var names = ['price', 'lo', 'la'];
    for(var i=0;i<names.length;i++){
        var s = params[names[i]];
        if(s && s.length>0){
            if(s.indexOf('-')==-1){
                return names[i];
            }
        }
    }
    return null;
}

Ext.onReady(loadit);